### ISSUE TITLE
Fix S1128 FP: documentation tags are not taken into account

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2694]
### Description

[Rule S1128: Unused "using" should be removed](https://rules.sonarsource.com/csharp/RSPEC-1128)

When checking which namespaces are used, documentation tags like `<see cref='SomeClass' />` are not taken into account.

### Repro steps
``` C#
using System; // Compliant, used by cref.

namespace SonarAnalyzer.Experiments.CSharp
{
    public enum S1128
    {
        /// <summary><see cref="DateTime"/></summary>
        DateTimeValue,
    }
}
```

### Known workarounds

``` C#
namespace SonarAnalyzer.Experiments.CSharp
{
    public enum S1128
    {
        /// <summary><see cref="System.DateTime"/></summary>
        DateTimeValue,
    }
}
```

### Related information
* SonarC# Version 7.17.0.9346

**Comment by andrei-epure-sonarsource** [comment]
Also repro from @DuAell in #2737

### Repro steps

1. Create an interface with a 'using'
```
using Framework.Exceptions;
```

2. Declare a method
```
/// <summary>
/// Start service. If service is already running, nothing happens.
/// </summary>
/// <exception cref="WindowsServiceException">thrown if service cannot be started.</exception>
void Start();
```
### Expected behavior

S1128 shouldn't be detected as there is no error there

### Actual behavior

A code smell is detected

**Comment by andrei-epure-sonarsource** [comment]
Another repro from @nnpcYvIVl 

```
using System.ComponentModel;

namespace S1128_False_Positive
{
    public class Class
    {
        /// <exception cref="Win32Exception">
        /// </exception>
        public void Test() => System.Console.WriteLine("Test");
    }
}
```

**Comment by andrei-epure-sonarsource** [comment]
another one from @jcurl , with `seealso`

```
using System.Collections.Generic;

/// <summary>
/// A linked list node for our block list.
/// </summary>
/// <remarks>
/// This class is not 100% compatible to <seealso cref="LinkedListNode{T}"/>.
/// </remarks>
public class MyClass {
  ...
}
```

**Comment by mgasparel** [comment]
Was able to reproduce this in a new project pulling the latest nuget package (8.13.1.21947). Is this a regression?

**Comment by pavel-mikula-sonarsource** [comment]
Hi @mgasparel,

Can you please share your reproducer code? I gave it quick try with 8.13.1 nuget, but it didn't reproduce.

**Comment by mgasparel** [comment]
@pavel-mikula-sonarsource 

I just used the original repro from the first post in this thread, but pushed a project for you to take a look: https://github.com/mgasparel/sonar-repro

Output:
```
🚀 dotnet build
Microsoft (R) Build Engine version 16.7.0-preview-20360-03+188921e2f for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
S1128.cs(1,1): warning S1128: Remove this unnecessary 'using'. [C:\Users\michaelgasparelli\code\sonartest\sonar-repro\sonar-repro.csproj]
  sonar-repro -> C:\Users\michaelgasparelli\code\sonartest\sonar-repro\bin\Debug\netcoreapp3.1\sonar-repro.dll

Build succeeded.

S1128.cs(1,1): warning S1128: Remove this unnecessary 'using'. [C:\Users\michaelgasparelli\code\sonartest\sonar-repro\sonar-repro.csproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.98
```

**Comment by pavel-mikula-sonarsource** [comment]
Ok, this is interesting.

`c.Node` is a compilation unit. On my machine, VS2019 16.7.3 `c.Node.DescendantTrivia()`:
* returns what's expected under VS
* returns empty collection under `dotnet build` command
* returns what's expected with `c.Node.DescendantTrivia(x => true)`

While other nodes (like `EnumDeclarationSyntax`) returns what's expected.

cc @andrei-epure-sonarsource, @costin-zaharia-sonarsource

**Comment by pavel-mikula-sonarsource** [comment]
This issue was closed and released as 8.1 milestone so I'll keep it closed. I've created a new #3633 to track the new problem.
